x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityIndexManager.java

```diff
@@ -361,10 +369,27 @@ public void onFailure(Exception e) {
     }
 
     private Tuple<String, Settings> loadMappingAndSettingsSourceFromTemplate() {
-        final byte[] template = TemplateUtils.loadTemplate("/" + SECURITY_TEMPLATE_NAME + ".json",
-                Version.CURRENT.toString(), SecurityIndexManager.TEMPLATE_VERSION_PATTERN).getBytes(StandardCharsets.UTF_8);
-        PutIndexTemplateRequest request = new PutIndexTemplateRequest(SECURITY_TEMPLATE_NAME).source(template, XContentType.JSON);
-        return new Tuple<>(request.mappings().get("doc"), request.settings());
+        final byte[] template = TemplateUtils.loadTemplate("/" + SECURITY_TEMPLATE_NAME + ".json", Version.CURRENT.toString(),
+                SecurityIndexManager.TEMPLATE_VERSION_PATTERN).getBytes(StandardCharsets.UTF_8);
+        final PutIndexTemplateRequest request = new PutIndexTemplateRequest(SECURITY_TEMPLATE_NAME).source(template, XContentType.JSON);
+
+        final String mappingSource = request.mappings().get(MapperService.SINGLE_MAPPING_NAME);
+        try (XContentParser parser = XContentType.JSON.xContent().createParser(NamedXContentRegistry.EMPTY,
+                DeprecationHandler.THROW_UNSUPPORTED_OPERATION, mappingSource)) {
+            // remove the type wrapping to get the mapping
+            if (parser.nextToken() == XContentParser.Token.START_OBJECT) { // {
+                if (parser.nextToken() == XContentParser.Token.FIELD_NAME) { // "_doc"
+                    if (parser.nextToken() == XContentParser.Token.START_OBJECT) { // {
+                        XContentBuilder builder = JsonXContent.contentBuilder();
+                        builder.generator().copyCurrentStructure(parser);
+                        return new Tuple<>(Strings.toString(builder), request.settings());
+                    }
+                }
+            }
+            throw new ElasticsearchException("cannot read mapping from security template");
```

Comment:
I recognise that this was pulled from my sample code, but I think we ought to provide more details in this error.
At the very least, what is the `currentToken` and `location` that caused a problem.

Alternatively we could switch those `nextToken` calls to use `XContentParserUtils.ensureExpectedToken` instead which will handle the exception for us.